MOB-21(Создание навигации)/MOB-22(Создание контейнера для фрагментов)

UBaHbI4 opened the pull request:
- Создание контейнера для фрагментов

UBaHbI4 committed:
MOB-22
- Создание контейнера для фрагментов

ialeksandrovi commented on app/src/main/res/layout/content_main.xml:
```diff
@@ -0,0 +1,19 @@
+    <fragment
+        android:id="@+id/nav_host_fragment_content_main"
+        android:name="androidx.navigation.fragment.NavHostFragment"
+        android:layout_width="match_parent"
```
чтобы содержимое ConstraintLayout занимало всю площадь родителя, лучше вместо match_parent использовать 0dp + конечно же, поместить это в ресурсы

ialeksandrovi commented on app/src/main/res/layout/content_main.xml:
```diff
@@ -0,0 +1,19 @@
+    tools:showIn="@layout/app_bar_main">
+
+    <fragment
+        android:id="@+id/nav_host_fragment_content_main"
```
На уровне предложения: мы на работе используем следующую методику, сначала инициалы вью, а затем его название. 

Ну и для контейнеров лучше использовать самое простое вью, FrameLayout

в итоге я бы предложил нейминг android:id="@+id/flContainer"

UBaHbI4 replied:
Может так и нагляднее, но нас вроде учили в xml использовать стиль с маленькими буквами и разделять нижним подчеркиванием. Типа frame_layout_container

ialeksandrovi commented on app/src/main/res/layout/nav_header_main.xml:
```diff
@@ -0,0 +1,91 @@
+
+    <ImageView
+        android:id="@+id/logo"
+        android:layout_width="@dimen/zero"
```
О, тем более что у тебя 0dp уже заведено в ресурсах! 
Круто!

ialeksandrovi commented on app/src/main/res/layout/nav_header_main.xml:
```diff
@@ -0,0 +1,91 @@
+            android:layout_marginVertical="@dimen/nav_header_title_margin_vertical"
+            android:paddingTop="@dimen/nav_header_vertical_spacing"
+            android:textAlignment="center"
+            android:textColor="@color/white"
```
а вот эти атрибуты лучше заведи сразу в стили.

UBaHbI4 replied:
Там Денис завел стили и цвета приложения, я после мержа их применю. Просто пока для меня они недоступны :)

ialeksandrovi commented on app/src/main/res/layout/nav_header_main.xml:
```diff
@@ -0,0 +1,91 @@
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/nav_header_user_margin_start"
+            android:textAlignment="textStart"
+            android:textColor="@color/white"
```
стиль

ialeksandrovi commented on app/src/main/res/layout/nav_header_main.xml:
```diff
@@ -0,0 +1,91 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/nav_header_user_margin_start"
+            android:textAlignment="textStart"
```
стиль

ialeksandrovi commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,8 @@
     <string name="app_name">Mobile Waiter</string>
+    <string name="nav_header_logo_description">Логотип</string>
+    <string name="nav_header_user_icon_description">Иконка пользователя</string>
+    <string name="nav_header_sample_company_title">Company title</string>
```
Лучше сразу заведи английский и русский словарики отдельно

UBaHbI4 committed:
Исправление замечаний:
- fragment заменил на FrameLayout, id переименовал, width и height выставил по 0dp через ресурсы
- Стили и цвета применю через один пулреквест, т.к. их уже добавил Денис в проект.
- Добавил strings для En локали.

UBaHbI4 commented:
Замечания исправил частично в 304fc444818640a1bc34fb8b7fa772e4c0315238. Про стили и цвета переделаю, после объединения с пулреквестом Дениса, который уже добавил цвета и стили для приложения, а так же иконки заменю.

UBaHbI4 committed:
Merge branch 'shivi/mob-21(create_navigation)/mob-21(create_navigation)' into shivi/mob-21(create_navigation)/mob-22(create_fragment_container)

GeologDen reviewed (approved):
Approve
